src/latbin/lattice-compose.cc

```diff
@@ -63,14 +75,30 @@ int main(int argc, char *argv[]) {
     }
 
     KALDI_ASSERT(phi_label > 0 || phi_label == fst::kNoLabel); // e.g. 0 not allowed.
+    KALDI_ASSERT(rho_label > 0 || rho_label == fst::kNoLabel); // e.g. 0 not allowed.
+    if (phi_label > 0 && rho_label > 0) {
+      KALDI_ERR << "You cannot set both 'phi_label' and 'rho_label' at the same time.";
+    }
+
+    { // convert 'compose_with_fst' to lowercase to support: true, True, TRUE
+      std::string tmp_lc(compose_with_fst);
+      std::transform(compose_with_fst.begin(), compose_with_fst.end(),
+                     tmp_lc.begin(), ::tolower);  // lc
```

Comment:
```suggestion
                     compose_with_fst.begin(), std::tolower);  // lc
```
It can be an **in-place** operation.